website/docs/api/kb.md

```diff
@@ -4,169 +4,88 @@ teaser:
   A storage class for entities and aliases of a specific knowledge base
   (ontology)
 tag: class
-source: spacy/kb.pyx
+source: spacy/kb/kb.pyx
 new: 2.2
 ---
 
-The `KnowledgeBase` object provides a method to generate
-[`Candidate`](/api/kb/#candidate) objects, which are plausible external
-identifiers given a certain textual mention. Each such `Candidate` holds
-information from the relevant KB entities, such as its frequency in text and
-possible aliases. Each entity in the knowledge base also has a pretrained entity
-vector of a fixed size.
+The `KnowledgeBase` object is an abstract class providing a method to generate
+[`Candidate`](/api/candidate) objects, which are plausible external identifiers
+given a certain textual mention. Each such `Candidate` holds information from
+the relevant KB entities, such as its frequency in text and possible aliases.
+Each entity in the knowledge base also has a pretrained entity vector of a fixed
+size.
+
+Beyond that, `KnowledgeBase` classes have to implement a number of utility
+functions called by the [`EntityLinker`](/api/entitylinker) component.
 
 ## KnowledgeBase.\_\_init\_\_ {#init tag="method"}
 
-Create the knowledge base.
+`KnowledgeBase` is an abstract class and cannot be instantiated. Its child
+classes should call `__init__()` to set up some necessary attributes.
 
 > #### Example
 >
 > ```python
 > from spacy.kb import KnowledgeBase
+> from spacy.vocab import Vocab
+>
+> class FullyImplementedKB(KnowledgeBase):
+>   def __init__(self, vocab: Vocab, entity_vector_length: int):
+>       super().__init__(vocab, entity_vector_length)
+>       ...
 > vocab = nlp.vocab
-> kb = KnowledgeBase(vocab=vocab, entity_vector_length=64)
+> kb = FullyImplementedKB(vocab=vocab, entity_vector_length=64)
 > ```
 
 | Name                   | Description                                      |
 | ---------------------- | ------------------------------------------------ |
 | `vocab`                | The shared vocabulary. ~~Vocab~~                 |
 | `entity_vector_length` | Length of the fixed-size entity vectors. ~~int~~ |
 
-## KnowledgeBase.entity_vector_length {#entity_vector_length tag="property"}
```

Comment:
This property is still available, no?

Reply:
True, deleted it by mistake. Added back in.